pandas/tests/arrays/test_datetimelike.py

```diff
@@ -868,3 +868,10 @@ def test_searchsorted_datetimelike_with_listlike_invalid_dtype(values, arg):
     msg = "[Unexpected type|Cannot compare]"
     with pytest.raises(TypeError, match=msg):
         values.searchsorted(arg)
+
+
+@pytest.mark.parametrize("klass", [list, tuple, np.array, pd.Series])
+def test_period_index_construction_from_strings(klass):
+    result = PeriodIndex(klass(["2020Q1", "2020Q2", "2020Q3", "2020Q4"]), freq="Q")
```

Comment:
can you do data = ...

so that result = statement is the bit being tested. (i.e. we are not testing the klass constructor)

can you also use the data from the OP, since that was not monotonic